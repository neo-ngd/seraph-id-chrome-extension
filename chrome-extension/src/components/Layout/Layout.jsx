// Copyright (c) 2019 Swisscom Blockchain AG
// Licensed under MIT License

import React from 'react';
import PropTypes from 'prop-types';
import { Box, CircularProgress } from '@material-ui/core';

/**
 * <Layout />
 * Base layout component.
 * It adds the flexbox container and handle the loading state.
 * @param children
 * @param justifyStart
 * @param padding
 * @param isLoading
 * @return {*}
 * @constructor
 */
const Layout = ({ children, justifyStart, padding, isLoading }) => {
  return (
    <Box
      justifyContent={justifyStart ? 'flex-start' : 'space-between'}
      display="flex"
      flexDirection="column"
      height="100%"
      flex={'0 0 100%'}
      padding={padding}
      overflow="hidden"
      maxWidth="100%"
    >
      {isLoading ? (
        <Box
          style={{
            margin: 'auto',
          }}
        >
          <CircularProgress style={{ color: '#00e599' }} />
        </Box>
      ) : (
        children
      )}
    </Box>
  );
};

Layout.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  justifyStart: PropTypes.bool,
  padding: PropTypes.string,
  isLoading: PropTypes.bool,
};

export default Layout;
