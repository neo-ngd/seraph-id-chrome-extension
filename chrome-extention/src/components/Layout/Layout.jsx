import React from 'react';
import PropTypes from 'prop-types';
import { Box, CircularProgress } from '@material-ui/core';

const Layout = ({ children, justifyStart, padding, isLoading }) => {
  return (
    <Box
      justifyContent={justifyStart ? 'flex-start' : 'space-between'}
      display="flex"
      flexDirection="column"
      height="100%"
      flex={'0 0 100%'}
      padding={padding}
      overflow="hidden"
      maxWidth="100%"
    >
      {isLoading ? (
        <Box style={{
          margin: 'auto',
        }}>
          <CircularProgress style={{ color: 'white' }} />
        </Box>
      ) : children }
    </Box>
  );
};

Layout.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]),
  justifyStart: PropTypes.bool,
  padding: PropTypes.string,
  isLoading: PropTypes.bool,
};

export default Layout;
