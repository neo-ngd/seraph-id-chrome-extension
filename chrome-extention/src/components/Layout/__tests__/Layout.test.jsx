import React from 'react';
import { shallow } from 'enzyme';
import Layout from "../Layout";

const Child = () => (<span>test</span>);

const props = {
    children: <Child />,
    justifyStart: true,
    padding: '2px',
    isLoading: false,
};

describe('<Layout />', () => {
    it('should render component', () => {
        const wrapper = shallow(<Layout {...props} />);
        expect(wrapper).toMatchSnapshot()
    });

    it('should render component with space-between prop', () => {
        const wrapper = shallow(<Layout {...props} justifyStart={false} />);
        expect(wrapper).toMatchSnapshot()
    });

    it('should render component in loading state', () => {
        const wrapper = shallow(<Layout {...props} isLoading={true} />);
        expect(wrapper).toMatchSnapshot()
    });
});
