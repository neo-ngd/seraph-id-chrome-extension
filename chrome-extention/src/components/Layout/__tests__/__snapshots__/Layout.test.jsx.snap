// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render component 1`] = `
<Styled(MuiBox)
  display="flex"
  flex="0 0 100%"
  flexDirection="column"
  height="100%"
  justifyContent="flex-start"
  maxWidth="100%"
  overflow="hidden"
  padding="2px"
>
  <Child />
</Styled(MuiBox)>
`;

exports[`<Layout /> should render component in loading state 1`] = `
<Styled(MuiBox)
  display="flex"
  flex="0 0 100%"
  flexDirection="column"
  height="100%"
  justifyContent="flex-start"
  maxWidth="100%"
  overflow="hidden"
  padding="2px"
>
  <Styled(MuiBox)
    style={
      Object {
        "margin": "auto",
      }
    }
  >
    <WithStyles(ForwardRef(CircularProgress))
      style={
        Object {
          "color": "white",
        }
      }
    />
  </Styled(MuiBox)>
</Styled(MuiBox)>
`;

exports[`<Layout /> should render component with space-between prop 1`] = `
<Styled(MuiBox)
  display="flex"
  flex="0 0 100%"
  flexDirection="column"
  height="100%"
  justifyContent="space-between"
  maxWidth="100%"
  overflow="hidden"
  padding="2px"
>
  <Child />
</Styled(MuiBox)>
`;
